lib/PuppeteerSharp/Tracing.cs

```diff
@@ -1,12 +1,117 @@
-﻿namespace PuppeteerSharp
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading.Tasks;
+using PuppeteerSharp.Messaging;
+
+namespace PuppeteerSharp
 {
     public class Tracing
     {
-        private Session client;
+        private Session _client;
+        private bool _recording;
+        private string _path;
+        private static readonly List<string> _defaultCategories = new List<string>()
+        {
+            "-*",
+            "devtools.timeline",
+            "v8.execute",
+            "disabled-by-default-devtools.timeline",
+            "disabled-by-default-devtools.timeline.frame",
+            "toplevel",
+            "blink.console",
+            "blink.user_timing",
+            "latencyInfo",
+            "disabled-by-default-devtools.timeline.stack",
+            "disabled-by-default-v8.cpu_profiler"
+        };
 
         public Tracing(Session client)
         {
-            this.client = client;
+            _client = client;
+        }
+
+        /// <summary>
+        /// Starts tracing.
+        /// </summary>
+        /// <returns>Start task</returns>
+        /// <param name="options">Tracing options</param>
+        public async Task StartAsync(TracingOptions options)
+        {
+            if (_recording)
+            {
+                Console.WriteLine("Cannot start recording trace while already recording trace.");

```

Comment:
I'm pretty sure that `console.assert` throws an error and not just logs, so I think we should throw an `InvalidOperationException`

Reply:
That's event clear to me, but yeah, let's treat that as a validation.